tycho-artifactcomparator/src/main/java/org/eclipse/tycho/zipcomparator/internal/TextContentsComparator.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 Hannes Wellmann and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.tycho.zipcomparator.internal;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.codehaus.plexus.component.annotations.Component;
+import org.eclipse.tycho.artifactcomparator.ArtifactComparator.ComparisonData;
+import org.eclipse.tycho.artifactcomparator.ArtifactDelta;
+import org.eclipse.tycho.artifactcomparator.ComparatorInputStream;
+
+@Component(role = ContentsComparator.class, hint = TextContentsComparator.TYPE)
+public class TextContentsComparator implements ContentsComparator {
+
+    public static final String DEFAULT_TEXT_FILE_EXTENSIONS = "txt,java,html,xml,exsd";
+
+    public static final String TYPE = "text";
+
+    private static final char CR = '\r';
+    private static final char LF = '\n';
+
+    @Override
+    public ArtifactDelta getDelta(ComparatorInputStream baselineStream, ComparatorInputStream reactorStream,
+            ComparisonData data) throws IOException {
+        byte[] baseline = baselineStream.asBytes();
+        byte[] reactor = reactorStream.asBytes();
+
+        int bI = 0;
+        int rI = 0;
+        int mismatch = Arrays.mismatch(baseline, reactor);
```

Comment:
This here seems a bit hard to understand if/how it works, why not having one for loop and simply advance the pointer if a newline is found? I'm also not sure if all cases are handled correctly I only see `\r\n` and `\n` handled but what about `\r` or even `\n\r`

and then I think this can be extracted into a public static method so it might be reused by other comparators.